codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AddBuiltinPlugins.java

```diff
@@ -74,10 +75,26 @@
                         .withConventions(AwsDependency.ADD_GLACIER_API_VERSION.dependency,
                                          "AddGlacierApiVersion", HAS_MIDDLEWARE)
                         .servicePredicate((m, s) -> testServiceId(s, "Glacier"))
+                        .build(),
+                RuntimeClientPlugin.builder()
+                        .withConventions(AwsDependency.BODY_CHECKSUM.dependency, "ApplyMd5BodyChecksum",
+                                         HAS_MIDDLEWARE)
+                        .servicePredicate((m, s) -> testServiceId(s, "S3"))
+                        .operationPredicate((m, s, o) -> s3Md5BodyChecksumOperations.contains(o.getId().getName()))
                         .build()
         );
     }
 
+    private static List<String> s3Md5BodyChecksumOperations = Arrays.asList(
```

Comment:
properties should appear before methods. Static properties appear before instance properties (public first, then package private I think, then protected, then private). This should also be a Set: `SetUtils.of(...)`.